server/src/event/model/set_taxonomy_parent.rs

```diff
@@ -28,3 +31,61 @@ impl TryFrom<&AbstractEvent> for SetTaxonomyParentEvent {
         })
     }
 }
+
+pub struct SetTaxonomyParentEventPayload {
+    raw_typename: RawEventType,
+    child_id: i32,
+    previous_parent_id: Option<i64>,
+    parent_id: i32,
+    actor_id: i32,
+    instance_id: i32,
+}
+
+impl SetTaxonomyParentEventPayload {
+    pub fn new(
+        child_id: i32,
+        previous_parent_id: Option<i64>,
+        parent_id: i32,
+        actor_id: i32,
+        instance_id: i32,
+    ) -> Self {
+        Self {
+            raw_typename: RawEventType::SetTaxonomyParent,
+            child_id,
+            previous_parent_id,
+            parent_id,
+            actor_id,
+            instance_id,
+        }
+    }
+
+    pub async fn save<'a, E>(&self, executor: E) -> Result<Event, EventError>
+    where
+        E: Executor<'a>,
+    {
+        let mut transaction = executor.begin().await?;
+
+        let mut uuid_parameters: HashMap<String, i32> = HashMap::new();
+
+        if let Some(previous_parent_id) = self.previous_parent_id {
+            uuid_parameters.insert("from".to_string(), previous_parent_id as i32);
+        }
+
+        uuid_parameters.insert("to".to_string(), self.parent_id);
+
+        let event = EventPayload::new(
+            self.raw_typename.clone(),
+            self.actor_id,
+            self.child_id,
+            self.instance_id,
+            HashMap::new(),
+            uuid_parameters,
+        )
+        .save(&mut transaction)
+        .await?;
+
+        transaction.commit().await?;
```

Comment:
```suggestion
        let mut uuid_parameters: HashMap<String, i32> = HashMap::new();

        if let Some(previous_parent_id) = self.previous_parent_id {
            uuid_parameters.insert("from".to_string(), previous_parent_id as i32);
        }

        uuid_parameters.insert("to".to_string(), self.parent_id);

        let event = EventPayload::new(
            self.raw_typename.clone(),
            self.actor_id,
            self.child_id,
            self.instance_id,
            HashMap::new(),
            uuid_parameters,
        )
        .save(executor)
        .await?;
```

Can you check in the other events that they are the same?

Reply:
I'm not sure what should be checked here. I suppose it i the use of `let mut uuid_parameters: HashMap<String, i32> = HashMap::new();` and so on. I've changed. Please take a look later.